pkg/provisioning/service.go

```diff
@@ -234,54 +237,30 @@ func (s *Service) collectConnectorStates(ctx context.Context, pl *pipeline.Insta
 			return nil, fmt.Errorf("could not get connector %q: %w", connID, err)
 		}
 
-		switch oldConn := oldConn.(type) {
-		case connector.Destination:
-			oldState := oldConn.State()
-			connStates[connID] = func(ctx context.Context) error {
-				newConn, err := s.connectorService.Get(ctx, connID)
-				if err != nil {
-					return fmt.Errorf("could not get connector %q: %w", connID, err)
-				}
-				if newConn.Config().Plugin != oldConn.Config().Plugin {
-					s.logger.Warn(ctx).
-						Str(log.PluginNameField, newConn.Config().Plugin).
-						Msg("plugin name changed, could not apply old destination state, the connector will start from the beginning")
-					return nil
-				}
-				_, err = s.connectorService.SetDestinationState(ctx, connID, oldState)
-				if cerrors.Is(err, connector.ErrInvalidConnectorType) {
-					s.logger.Warn(ctx).
-						Str(log.ConnectorIDField, connID).
-						Err(err).
-						Msg("could not apply old destination state, the connector will start from the beginning")
-					return nil
-				}
-				return err
+		oldState := oldConn.State
+		if oldState == nil {
+			continue // no need to copy state
+		}
+		connStates[connID] = func(ctx context.Context) error {
```

Comment:
Nitpick: would be nice to add a comment explaining what the function does (appears to be copying the connector state).